streams/src/test/java/org/apache/kafka/streams/state/internals/AbstractWindowBytesStoreTest.java

```diff
@@ -287,17 +308,43 @@ public void shouldFetchAllInTimeRange() {
         final KeyValue<Windowed<Integer>, String> four = windowedPair(4, "four", startTime + 4);
         final KeyValue<Windowed<Integer>, String> five = windowedPair(5, "five", startTime + 5);
 
-        assertEquals(
-            new HashSet<>(asList(one, two, four)),
-            toSet(windowStore.fetchAll(ofEpochMilli(startTime + 1), ofEpochMilli(startTime + 4)))
+        assertArrayEquals(
+            new LinkedHashSet<>(asList(one, two, four)).toArray(),
+            toSet(windowStore.fetchAll(ofEpochMilli(startTime + 1), ofEpochMilli(startTime + 4))).toArray()
         );
-        assertEquals(
-            new HashSet<>(asList(zero, one, two)),
-            toSet(windowStore.fetchAll(ofEpochMilli(startTime + 0), ofEpochMilli(startTime + 3)))
+        assertArrayEquals(
+            new LinkedHashSet<>(asList(zero, one, two)).toArray(),
+            toSet(windowStore.fetchAll(ofEpochMilli(startTime + 0), ofEpochMilli(startTime + 3))).toArray()
         );
-        assertEquals(
-            new HashSet<>(asList(one, two, four, five)),
-            toSet(windowStore.fetchAll(ofEpochMilli(startTime + 1), ofEpochMilli(startTime + 5)))
+        assertArrayEquals(
+            new LinkedHashSet<>(asList(one, two, four, five)).toArray(),
```

Comment:
It feels kind of ridiculous to convert this from a list to a set to an array all in one line. Maybe we can use `assertThat(result, equalTo(expectedResult))` here like we've started to do elsewhere in Streams?

Reply:
Wow this is embarrasing haha, looks like I was trying to get tests green without much thinking 😅

Reply:
Haha actually I think I was the one who added the initial conversion from list to set to begin with, so I was clearly asking for trouble